src/webgpu/gpu_test.ts

```diff
@@ -258,6 +250,44 @@ export class GPUTest extends Fixture {
     await this.selectDeviceOrSkipTestCase(features);
   }
 
+  /**
+   * Expects that the device should be lost for a particular reason at the teardown of the test.
+   */
+  expectDeviceLost(reason: GPUDeviceLostReason): void {
+    assert(
+      this.provider !== undefined,
+      'No provider available right now; did you "await" selectDeviceOrSkipTestCase?'
```

Comment:
It occurs to me that if we require all selectDeviceOrSkipTestCase calls to be in the before() function, it could behave like "eventual expectations" in that you'd no longer have to await them, and we can await automatically after before() runs.
In fact, the device "select" mechanism (and the device pool "reserve" mechanism) might get significantly simpler overall because we probably don't have to initialize a GPUDevice at all until after before() completes.

It *also* occurs to me that we may no longer technically need SkipTestCase to be throwable from the test function. I wonder if we could limit it only to before() functions?

Both of these can be followups, maybe leave MAINTENANCE_TODOs if you think they're worth considering.

Reply:
I added a todo for device selection. I think SkipTestCase is still useful for skipping subcases that aren't implemented or something like that

Reply:
FYI I started looking at this